eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Bbody.hs

```diff
@@ -142,27 +133,81 @@ instance
+          TooManyExUnits mm -> Sum TooManyExUnits 3 !> To mm
+          BodyRefScriptsSizeTooBig mm -> Sum BodyRefScriptsSizeTooBig 4 !> To mm
+      beforeEleven =
+        encode . \case
+          WrongBlockBodySizeBBODY mm@(Mismatch _ _) ->
+            Sum WrongBlockBodySizeBBODY 0
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
+          InvalidBodyHashBBODY mm@(Mismatch _ _) ->
+            Sum (InvalidBodyHashBBODY @era) 1
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
+          LedgersFailure x -> Sum (LedgersFailure @era) 2 !> To x
+          TooManyExUnits mm@(Mismatch _ _) ->
+            Sum TooManyExUnits 3
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
+          BodyRefScriptsSizeTooBig mm@(Mismatch _ _) ->
+            Sum BodyRefScriptsSizeTooBig 4
+              !> E (\(Mismatch {..}) -> encCBOR mismatchSupplied <> encCBOR mismatchExpected) mm
 
 instance
   ( Era era
   , DecCBOR (PredicateFailure (EraRule "LEDGERS" era))
   ) =>
   DecCBOR (ConwayBbodyPredFailure era)
   where
-  decCBOR = decode . Summands "ConwayBbodyPred" $ \case
-    0 -> SumD WrongBlockBodySizeBBODY <! From <! From
-    1 -> SumD InvalidBodyHashBBODY <! From <! From
-    2 -> SumD LedgersFailure <! From
-    3 -> SumD TooManyExUnits <! From <! From
-    4 -> SumD BodyRefScriptsSizeTooBig <! From <! From
-    n -> Invalid n
+  decCBOR = ifDecoderVersionAtLeast (natVersion @11) elevenOnwards beforeEleven
+    where
+      elevenOnwards = decode . Summands "ConwayBbodyPred" $ \case
+        0 -> SumD WrongBlockBodySizeBBODY <! From
+        1 -> SumD InvalidBodyHashBBODY <! From
+        2 -> SumD LedgersFailure <! From
+        3 -> SumD TooManyExUnits <! From
+        4 -> SumD BodyRefScriptsSizeTooBig <! From
+        n -> Invalid n
+      beforeEleven = decode . Summands "ConwayBbodyPred" $ \case
+        0 ->
+          SumD WrongBlockBodySizeBBODY
+            <! D
+              ( do
+                  mismatchSupplied <- decCBOR
+                  mismatchExpected <- decCBOR
+                  pure Mismatch {..}
+              )
+        1 ->
+          SumD InvalidBodyHashBBODY
+            <! D
+              ( do
+                  mismatchSupplied <- decCBOR
+                  mismatchExpected <- decCBOR
+                  pure Mismatch {..}
+              )
+        2 -> SumD LedgersFailure <! From
+        3 ->
+          SumD TooManyExUnits
+            <! D
+              ( do
+                  mismatchSupplied <- decCBOR
+                  mismatchExpected <- decCBOR
+                  pure Mismatch {..}
+              )
+        4 ->
+          SumD BodyRefScriptsSizeTooBig
+            <! D
+              ( do
+                  mismatchSupplied <- decCBOR
+                  mismatchExpected <- decCBOR
+                  pure Mismatch {..}
+              )
+        n -> Invalid n
```

Comment:
As I mentioned during the meeting. We cannot change serialization for protocol versions that are used for intra-era hard forks. It is unknown yet if `11` will be a new era or an intra era hard fork, so it is not safe to use it that way. Moreover, if we are going to be changing a predicate failure serialization for a new era, we might as well redefine it as a new predicate failure in the new era.

Moreover, `DecCBORGRoup` instance will transform this to:
```suggestion
  decCBOR = decode . Summands "ConwayBbodyPred" $ \case
    0 -> SumD WrongBlockBodySizeBBODY <! D decCBORGroup
    1 -> SumD InvalidBodyHashBBODY <! D decCBORGroup
    2 -> SumD LedgersFailure <! From
    3 -> SumD TooManyExUnits <! D decCBORGroup
    4 -> SumD BodyRefScriptsSizeTooBig <! D decCBORGroup
    n -> Invalid n
```

Reply:
I have removed all checks on the protocol version. 👍